contracts/adapters/TributeNFT.sol

```diff
@@ -0,0 +1,267 @@
+SOFTWARE.
+ */
+
+contract TributeNFTContract is
+    ITribute,
+    DaoConstants,
+    MemberGuard,
+    AdapterGuard
+{
+    using Address for address;
+
+    struct ProposalDetails {
+        // The proposal id.
+        bytes32 id;
+        // The applicant address (who will receive the DAO Shares and become a member).
+        address applicant;
+        // The address of the NFT to be locked in the DAO in exchange for voting power.
+        address nftAddr;
+        // The nft token identifier.
+        uint256 nftTokenId;
+        // The amount of shares requested to lock the NFT in exchange of DAO internal tokens.
+        uint256 requestedShares;
+    }
+
+    // Keeps track of all nft tribute proposals handled by each DAO.
+    mapping(address => mapping(bytes32 => ProposalDetails)) public proposals;
+
+    /**
+     * @notice default fallback function to prevent from sending ether to the contract.
+     */
+    receive() external payable {
+        revert("fallback revert");
+    }
+
+    function configureDao(DaoRegistry dao, address nftToCollect)
+        external
+        onlyAdapter(dao)
+    {
+        NFTExtension nftExt = NFTExtension(dao.getExtensionAddress(NFT));
+        nftExt.registerPotentialNewNFT(nftToCollect);
+    }
+
+    function provideTribute(
+        DaoRegistry dao,
+        bytes32 proposalId,
+        address applicant,
+        address tokenToMint,
+        uint256 requestAmount,
+        address tokenAddr,
+        uint256 tributeAmount
+    ) external pure override {
+        revert("not supported operation");
+    }
+
+    /**
+     * @notice Creates a tribute proposal and escrows received tokens into the adapter.
+     * @dev Applicant address must not be reserved.
+     * @dev The proposer must first separately `approve` the adapter as spender of the ERC-20 tokens provided as tribute.
+     * @param dao The DAO address.
+     * @param proposalId The proposal id (managed by the client).
+     * @param nftAddr The address of the ERC-721 NFT DAO that will be locked in the DAO in exchange for Shares.
+     * @param nftTokenId The NFT token id.
+     * @param requestedShares The amount of Shares requested of DAO as voting power.
+     */
+    function provideNFTTribute(
+        DaoRegistry dao,
+        bytes32 proposalId,
+        address nftAddr,
+        uint256 nftTokenId,
+        uint256 requestedShares
+    ) external override {
+        address applicant = msg.sender;
+        require(
+            dao.isNotReservedAddress(applicant),
+            "applicant is reserved address"
+        );
+
+        require(
+            !dao.isActiveMember(applicant),
+            "applicant is already a member"
+        );
+
+        // Transfers the NFT to the DAO Collection, and checks if the NFT is supported/valid.
+        NFTExtension nftExt = NFTExtension(dao.getExtensionAddress(NFT));
+        nftExt.transferFrom(applicant, nftAddr, nftTokenId);
```

Comment:
I guess transferFrom is to transfer the NFT from the applicant to the address right ? we should do that to the adapter first.

Then we send the NFT from the adapter to the NFTExtension once the proposal is being processed

Reply:
Ah ok, that makes sense. I was always sending to the NFTExtension, regardless of the proposal status.